server/api.go

```diff
@@ -184,10 +184,11 @@ func fillInContext(context middleware.Context, db db.DB,
 	r *http.Request, w http.ResponseWriter,
 	s *schema.Schema, p martini.Params, sync sync.Sync,
 	identityService middleware.IdentityService,
-	queue *job.Queue) {
+	queue *job.Queue, requestData map[string]interface{}) {
 	context["path"] = r.URL.Path
 	context["http_request"] = r
 	context["http_response"] = w
+	context["request_data"] = requestData
```

Comment:
[there are places](https://github.com/cloudwan/gohan/blob/master/server/resources/resource_management.go#L773) where the dataMap is modified, so the content of `context["request_data"]` may be modified after initial assignment. is it OK?
if not, we'd need a deep copy

Reply:
Deepcopy added